lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -2939,9 +2939,13 @@ public function addSqlResultSetMapping(array $resultMapping)
                             if (str_contains($fieldName, '.')) {
                                 [, $fieldName] = explode('.', $fieldName);
                             }
-
                             $resultMapping['entities'][$key]['fields'][$k]['column'] = $fieldName;
                         }
+                        $column = $resultMapping['entities'][$key]['fields'][$k]['column'];
+                        if ($column[0] === '`') {
```

Comment:
This looks quite hacky. Is it the only place where it's necessary? I assume it's only added here because the corresponding test uses a keyword. Is that test supposed to test a keyword or it happens to be one on Oracle?

Note that there is `Doctrine\DBAL\Schema\Identifier` which implements some similar logic.

Reply:
As a matter of fact only for the reserved words the decision. The `ClassMetadataInfo` class has this check in many places. We need to look at `Doctrine\DBAL\Schema\Identifier `and rewrite if this is what we need

Reply:
`Doctrine\DBAL\Schema\Identifier` is not suitable for solving the problem